### DISCUSSION TITLE
Request BrowserId : 1 not found it's likely the browser is already closed

### DISCUSSION THREAD

**Discussion by sxsean1** [cefsharp/CefSharp#4606] [General]
private async void _browser_LoadingStateChanged2(object sender, LoadingStateChangedEventArgs e)
    {
        if (!e.IsLoading)
        {
            JavascriptResponse resp = await _browser.GetMainFrame().EvaluateScriptAsync(
                @"document.getElementsByClassName('xt0psk2')[0]['href'];"
            );
        }
    }
110版本可以  新版本提示
:EvaluateScriptAsyncChromiumWebBrowserRequest BrowserId : 1 not found it's likely the browser is already closedloadingStateChanged

**Comment by amaitland** [comment]
English please.

Please provide more information, answer questions at https://github.com/cefsharp/CefSharp/wiki/Questions-&-Support#when-asking-a-question-please-include-the-following

**Comment by sxsean1** [comment]
private async void _browser_LoadingStateChanged2(object sender, LoadingStateChangedEventArgs e)
{
	if (!e.IsLoading)
	{
	JavascriptResponse resp = await _browser.GetMainFrame().EvaluateScriptAsync(
	@"document.getElementsByClassName('xt0psk2')[0]['href'];"
	);
	}
}
I sometimes encounter this mistake for no reason. (EvaluateScriptAsyncChromiumWebBrowserRequest BrowserId: 1 not found it's like the browser is already closed)

I think this is an error in Cefs browser (115- until the latest version -117.2.40). After multiple page jumps, sometimes it becomes invalid
I found that analogs of this code can work properly
Browser ExecuteScriptAsync (script);

https://ru.stackoverflow.com/questions/1534819/request-browserid-1-not-found-its-likely-the-browser-is-already-closed

**Reply by amaitland** [reply]
Your welcome to review the most recent changes see https://github.com/cefsharp/CefSharp/compare/v115.3.130...v117.2.40

More likely there's been a behaviour change in `Chromium`.

**Comment by ficho-hun** [comment]
That happened me too with the new versions.
After the page reloaded because of navigation or redirection, every script execution failed.
I tried the Browser.ExecuteAsync(script) as a workaround, but since that using the same functions that produce the same error.

**Comment by amaitland** [comment]
An issue has been created https://github.com/cefsharp/CefSharp/issues/4621

Moving the discussion there.

**Reply by amaitland** [reply]
If someone can provide a failing unit tests that reproduces the problem then see https://github.com/cefsharp/CefSharp/issues/4621#issuecomment-1802991766
